src/toolkitAPI.c

```diff
@@ -951,6 +974,36 @@ int DLLEXPORT swmm_getSubcatchResult(int index, int type, double *result)
     return(errcode);
 }
 
+int DLLEXPORT swmm_getGagePrecip(int index, double *rainfall, double *snowfall, double *total)
+//
```

Comment:
The API would be cleaner is you returned an array of three doubles instead of three individual doubles.

Reply:
@michaeltryby, I feel it works either way.  For discrete values, I am a fan of having separate variables.  Then I can just use what I need.  There is a tipping point, however. If we have 10 items that are served to the user, it doesn't make sense to handle the separately.  IMO, the future of this function really isn't going to expand beyond these three parameters. Unless maybe SWMM starts modeling raining "cats" and "dogs".  I feel that this will meet our use cases best.

Reply:
@michaeltryby, yes I see more clearly your suggestion.  I can make this change tonight.  

It might be a good idea to raise an issue on API style and we can talk discuss the best approach for scooping data out of SWMM

Reply:
@abhiramm7, @michaeltryby is suggesting that instead of passing three discrete values by reference that we return an array of three elements. You sculpted this function so what do you think about Michael’s suggestion?

Reply:
@bemcdonnell I agree with @michaeltryby suggestion. I can submit a PR to that extent.

Reply:
@bemcdonnell I agree with you on the pushing the API style changes to a separate PR. I think we need to fix other functions too. For now I suggest we go ahead with these changes.

Reply:
@abhiramm7, I’ll do it when I make the final edits